### ISSUE TITLE
关于训练的问题

### ISSUE THREAD

**Issue by dreamchaser-cai** [jingtianyilong/YOLOv4-pytorch#8]
您好可以加一下您的联系方式吗，我有一些关于这个项目训练的问题想请教一下您。

**Comment by jingtianyilong** [comment]
Please ask in the issue or discussion. Because I believe that your question might also be ask by others. And by answering your question, we can avoid answering same question again and again.
Discussion section is also available for this project. I will check them every now and then.

**Comment by dreamchaser-cai** [comment]
ok，I have a problem when training this project with my own data.My data format is like this：
- data
  - people
    - images
    - labels
  - test_annotation.txt
  - train_annotation.txt
  
The txt file stored the path of the picture。then，i meet a proplem like this
`IndexError: too many indices for array`
I'm not sure if my data format is wrong，Thanks for your answering。

**Comment by jingtianyilong** [comment]
Well. Make sure you read the `README.md` before you do anything.
# How to use
## Dataset
The code had been heavily modified, so the COCO and VOC format are not supported currently. So pls convert your dataset from whatever format into this specific yolo format. We use text file which describing the path to the image and its bboxes.
Each line would contain only one image and all the annotation about the bbox comes right after the image path. See example below:
```
images/00001.png x1,y1,x2,y2,class_id x1,y1,x2,y2,class_id x1,y1,x2,y2,class_id x1,y1,x2,y2,class_id 
images/00002.png x1,y1,x2,y2,class_id x1,y1,x2,y2,class_id 
```
`x1, y1` refer to the coordinates of top left corner.
`x2, y2` refer to the coordinates of bottom right corner.
All the coordinates are pixel wise location on the original image.

To make sure that the COCOAPI works properly, you might also have to change your image name to a number e.g. `0000234.png`. Or else you might modify the `get_image_id` in `utils/datasets.py` and other function related to index respectively.

########################################################################

Also, I can't really answer your question when you only tell me that you got `IndexError`. Because it can be anything. Spot the line of code that dump the error and see if you can actually do something. What is this `list`?  What does it contains? Do they have any meaning? Are they the thing you wish to have? If not, then why not? Which line modifies the `list`? What did it put in? You can normally source back to where the problem actually occur.

**Comment by ogbanugot** [comment]
![Screenshot from 2021-06-25 18-18-18](https://user-images.githubusercontent.com/24531935/123462176-def6b480-d5e1-11eb-8c1f-770f95ebfecc.png)
Having the same issue; and my train.txt file is formatted correctly. Any idea why this is happening please?

**Comment by ogbanugot** [comment]
Yeah, so the problem was that some images had no bounding box annotations. So you might see images with blank spaces (highlighted in red in the image attached) in your "train.txt" file. Just remove those images and the error will go away.
![Screenshot from 2021-06-25 22-46-43](https://user-images.githubusercontent.com/24531935/123488424-d44f1600-d607-11eb-9ac2-322629395ff4.png)
